cocos/asset/assets/simple-texture.ts

```diff
@@ -140,11 +142,11 @@ export class SimpleTexture extends TextureBase {
      * @param level @en Mipmap level to upload the image to. @zh 要上传的 mipmap 层级。
      * @param arrayIndex @en The array index. @zh 要上传的数组索引。
      */
-    public uploadData (source: HTMLCanvasElement | HTMLImageElement | ArrayBufferView | ImageBitmap, level = 0, arrayIndex = 0): void {
+    public uploadData (source: HTMLCanvasElement | HTMLImageElement | ArrayBufferView | ImageBitmap | IMemoryImageSource, level = 0, arrayIndex = 0): void {
```

Comment:
Can avoid the usage of `HTMLCanvasElement ` and other web standard API here?

Reply:
ImageSource is not exported and is also the internal interface of the engine. The interface for exporting PAL also needs to be considered

Reply:
Why can not export ImageSource?

Reply:
It seems that the export interface can also rely on internal types. I will reconsider this